sql/plan/showtablestatus.go

```diff
@@ -65,44 +61,60 @@ func (s *ShowTableStatus) Schema() sql.Schema { return showTableStatusSchema }
 
 // RowIter implements the sql.Node interface.
 func (s *ShowTableStatus) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
-	var rows []sql.Row
-	var tables []string
-	var err error
-	if len(s.Databases) > 0 {
-		for _, db := range s.Catalog.AllDatabases() {
-			if !stringContains(s.Databases, db.Name()) {
-				continue
-			}
+	dbName := s.Database
+	if dbName == "" {
+		return nil, sql.ErrNoDatabaseSelected.New()
+	}
 
-			tables, err = db.GetTableNames(ctx)
+	db, err := s.Catalog.Database(dbName)
```

Comment:
Implement sql.Databaser and set it in the analzyer, rather than getting it at runtime